serial.go

```diff
@@ -84,12 +89,36 @@ type Config struct {
 	// XONFlowControl bool
 
 	// CRLFTranslate bool
-	// TimeoutStuff int
 }
 
 // OpenPort opens a serial port with the specified configuration
 func OpenPort(c *Config) (io.ReadWriteCloser, error) {
-	return openPort(c.Name, c.Baud)
+	return openPort(c.Name, c.Baud, c.ReadTimeout)
+}
+
+// Converts the timeout values for Linux / POSIX systems
+func posixTimeoutValues(readTimeout time.Duration) (vmin uint8, vtime uint8) {
+	const MAXUINT8 = 1<<8 - 1 // 255
+	// set blocking / non-blocking read
+	var minBytesToRead uint8 = 1
+	var readTimeoutInDeci uint8 = 0
+	if readTimeout > 0 {
+		// EOF on zero read
+		minBytesToRead = 0
+		timeoutMs := uint32(readTimeout.Nanoseconds() / 1e6)
```

Comment:
Why convert to a uint32?  That could cause wrapping.  I would leave it as an int64 like .Nanoseconds returns.  The bounds checks you have will work just the same.